src/WebJobs.Script/Utility.cs

```diff
@@ -64,6 +64,49 @@ internal static async Task InvokeWithRetriesAsync(Func<Task> action, int maxRetr
             }
         }
 
+        /// <summary>
+        /// Invokes the specified async action with retry.
+        /// </summary>
+        /// <typeparam name="TS">Input</typeparam>
+        /// <typeparam name="TR">Output</typeparam>
+        /// <param name="action">Async action to invoke</param>
+        /// <param name="source">Input factory</param>
+        /// <param name="successFunc">Function to check for action success</param>
+        /// <param name="msg">Log details</param>
+        /// <param name="logger">Optional logger</param>
+        /// <param name="maxRetries">Max number of retries until success</param>
+        /// <param name="delay">Delay between successive retries</param>
+        /// <returns>TR</returns>
+        internal static async Task<TR> InvokeAsyncActionWithRetry<TS, TR>(Func<TS, Task<TR>> action, Func<TS> source, Func<TR, bool> successFunc, string msg, ILogger logger, int maxRetries, TimeSpan delay)
```

Comment:
Why isn't the existing InvokeWithRetries method sufficuent? It already supports a delay interval, etc.

Reply:
The existing one handles exceptions differently. reuses initially captured variables which doesnt work for HttpRequests

Reply:
Fixed.